src/main/java/com/google/api/generator/gapic/composer/ServiceClientSampleCodeComposer.java

```diff
@@ -194,77 +195,107 @@ public static String composeRpcMethodHeaderSampleCode(
       List<MethodArgument> arguments,
       Map<String, ResourceName> resourceNames,
       Map<String, Message> messageTypes) {
-    // TODO(summerji): Add other types RPC methods' sample code.
     if (method.isPaged()) {
-      // Find the repeated field.
-      Message methodOutputMessage = messageTypes.get(method.outputType().reference().simpleName());
-      Field repeatedPagedResultsField = methodOutputMessage.findAndUnwrapFirstRepeatedField();
-      Preconditions.checkNotNull(
-          repeatedPagedResultsField,
-          String.format(
-              "No repeated field found on message %s for method %s",
-              methodOutputMessage.name(), method.name()));
-
-      TypeNode repeatedResponseType = repeatedPagedResultsField.type();
       return SampleCodeWriter.write(
           composeUnaryPagedRpcMethodSampleCode(
-              method, clientType, repeatedResponseType, arguments, resourceNames));
+              method, clientType, arguments, resourceNames, messageTypes, false));
     }
     if (method.hasLro()) {
       return SampleCodeWriter.write(
-          composeUnaryLroRpcMethodSampleCode(method, clientType, arguments, resourceNames));
+          composeUnaryLroRpcMethodSampleCode(
+              method, clientType, arguments, resourceNames, messageTypes, false));
     }
     return SampleCodeWriter.write(
-        composeUnaryRpcMethodSampleCode(method, clientType, arguments, resourceNames));
+        composeUnaryRpcMethodSampleCode(
+            method, clientType, arguments, resourceNames, messageTypes, false));
   }
 
-  @VisibleForTesting
-  static TryCatchStatement composeUnaryRpcMethodSampleCode(
+  public static String composeRpcDefaultMethodHeaderSampleCode(
+      Method method,
+      TypeNode clientType,
+      Map<String, ResourceName> resourceNames,
+      Map<String, Message> messageTypes) {
+    if (method.isPaged()) {
+      return SampleCodeWriter.write(
+          composeUnaryPagedRpcMethodSampleCode(
+              method, clientType, Collections.emptyList(), resourceNames, messageTypes, true));
```

Comment:
The method-specific sample code composer should not know anything about whether they are generating a default sample - that responsibility belongs to this method (`composeRpcDefault`). Instead, this method should generate the `requestVarExpr`, and we can refactor the helper method as follows (using `composeUnaryPagedRpcMethodSampleCode` as an example):
```
composeUnaryRpcMethodSampleCode(
      Method method,
      TypeNode clientType,
      List<MethodArgument> arguments,
      Map<String, ResourceName> resourceNames, 
      List<Expr> bodyExprs,
      List<VariableExpr> rpcMethodArgVarExprs)
      
composeUnaryRpcMethodSampleCode(
      Method method,
      TypeNode clientType,
      List<MethodArgument> arguments,
      Map<String, ResourceName> resourceNames) {
    // Calls the method above.
}
```

Reply:
Agree that `composeRpcDefault` is responsible to generate `requestVarExpr`. I refactor the helper method to generate the body statements instead of TryCatchStatement as whole. Please take another look. Thanks.